# Error handling
{id: error-handling}


## Runtime error (panic) noticed during compilation.
{id: runtime-panic-noticed-during-compilation}

![](examples/errors/div_by_zero_hard_coded.rs)
![](examples/errors/div_by_zero_hard_coded.out)

## Divide by zero panick in function
{id: panick-in-function}

![](examples/errors/divide_by_zero_panick.rs)

## Return error on failure
{id: return-error-on-failure}

![](examples/errors/divide_by_zero_return_error.rs)


## Divide by zero runtime panic
{id: divide-by-zero-runtime-panic}

![](examples/errors/div_by_zero.rs)
![](examples/errors/div_by_zero.out)

## Divide by zero return error
{id: divide-by-zero-return-error}

![](examples/errors/div_by_zero_catch_error.rs)

## factorial function, runtime panic
{id: factorial-function-runtime-panic}

![](examples/errors/factorial.rs)
![](examples/errors/factorial.out)

## factorial create panic
{id: factorial-create-panic}

![](examples/errors/factorial_create_panic.rs)
![](examples/errors/factorial_create_panic.out)

## Out of bound for vectors
{id: out-of-bound-for-vectors}

* Run-time panic

![](examples/errors/out_of_bounds_vector.rs)
![](examples/errors/out_of_bounds_vector.out)

## Out of bound for arrays
{id: out-of-bound-for-arrays}

* Compile-time panic

![](examples/errors/out_of_bounds_array.rs)
![](examples/errors/out_of_bounds_array.out)

## Open files
{id: open-files}

![](examples/errors/file_open.rs)

